utils/CMakeLists.txt

```diff
@@ -1,5 +1,16 @@
 include_directories("../libproxy")
 
 add_executable(proxy proxy.c)
-target_link_libraries(proxy libproxy)
+
+if(BUILD_SHARED_LIBS)
+  target_link_libraries(proxy libproxy)
+else ()
+  if(WIN32)
+    target_link_libraries(proxy libproxy;modman;ws2_32;${LIBPROXY_LIBRARIES})
```

Comment:
If the dependencies of the `libproxy` lib are set up properly, this should not be necessary.

This might be broken because of those lines in `libproxy.cmk`:

```
set_target_properties(libproxy PROPERTIES INTERFACE_LINK_LIBRARIES "")
set_target_properties(libproxy PROPERTIES LINK_INTERFACE_LIBRARIES "")
```

Reply:
Do you have an idea on how to fix it?

Reply:
Yes, try to build it without those lines and then just use `target_link_libraries(proxy libproxy)` in all cases.

Reply:
If I do that then proxy won't have the pacrunner static libraries that's why I'm using `${LIBPROXY_LIBRARIES}` so it picks them up. I guess that for each library I could call `target_link_libraries(proxy ${PACRUNNER_LIB})` but that would require adding that line in multiple cmake files and doesn't make sense to compile an example code.

The way I did it its cleaner IMO since I'm not adding code in multiple places just to make a "test code" to build (`proxy`).

Reply:
mmmm... But you need that if want to use a module to run pacrunner like ducktape. You can always uncheck all the runner modules but that's not the point of the proxy isn't it? I'm not very familiar with the proxy

Reply:
If I remove that I get a bunch of linker errors with `proxy`
<img width="554" alt="image" src="https://user-images.githubusercontent.com/5193128/197793119-0c434c6c-01e4-4d73-83fa-956f57d2e625.png">

Reply:
Which means that libproxy does not link against duktape correctly. I'll try to reproduce that here.

Reply:
which they won't appear if I don't select any pacrunner since it won't try statically link it into `proxy`.
<img width="347" alt="image" src="https://user-images.githubusercontent.com/5193128/197793799-3560a865-656f-4bdf-98d2-834c269925d2.png">

I think my code handles as nicer as possible the linking with the pacrunners that the user selects it wants to use in CMake but open to suggestions

Reply:
I wonder whether there is some windows specific bug somewhere. Here on Linux I can build `proxy` which includes libproxy statically but links to `duktape` dynamically just fine without that line.

Reply:
"but links to `duktape` dynamically". You don't want, at least I don't, to have `ducktape` linked dynamically. My version of ducktape is statically compiled and therefore it tries to statically compile `ducktape` into `proxy`.

I guess you can compile statically `libproxy` but not `ducktape` but that may not be that usefule (either you want it completely static or not static at all right?)

Reply:
> "but links to duktape dynamically". You don't want, at least I don't, to have ducktape linked dynamically. My version of ducktape is statically compiled and therefore it tries to statically compile ducktape into proxy.

In this case I only have dynamically built duktape here. That `proxy` was linked dynamically to duktape even though CMakeLists only specifies that it links to `libproxy` shows that the transitive dependencies are carried along as they should, and there's no need to do something like `target_link_libraries(proxy duktape)` in the static case.

Reply:
I changed the code back according to what you said but the problem persist. The `proxy` project only has `libproxy.lib` as dependency and not `modman`, `duktape`,  `ws2_32`, ...
<img width="164" alt="image" src="https://user-images.githubusercontent.com/5193128/198041048-4512f946-0cd4-4864-b99d-916e5e3d6ea0.png">

I understand your point that `target_link_libraries(proxy libproxy)` should carry out the dependencies that `libproxy` has but for some reason it is not. I'm running out of ideas. Can you check it out?

Thanks!!

Reply:
I found [this](https://stackoverflow.com/a/53673552/1618465):
```
ote again, that such linkage propagation for static libraries works only when myLib2 is used in the same project which calls target_link_libraries(myLib2 myLib1).

Attempt to target_link_libraries(myExe myLib2) from another project will link just with myLib2.a file, which doesn't contain information about myLib2.
```

So I searched in the entire repo for `project(` and it turns out that there are two `project(libproxy)`:
- CMakeLists.txt
- libproxy\CMakeLists.txt

If I comment the `project(libproxy)` from the first one it works fine and the dependencies get populated into `proxy` but not if I do it the other way around (which would make more sense to have the `project(libproxy)` in the root `CMakelists.txt` (actually CMake complains when generating the project).

Thoughts?

Reply:
Yeah, there should be just a single `project` call in the entire repository.

Reply:
any idea why leaving just the one in the root folder doesn't make it spread the dependencies properly?

Reply:
Maybe there's some global property which this resets?

Reply:
I've been trying to find out why that happens but I've run out of ideas. Can you please take a look at it?

Reply:
@Vogtinator can you give me a hand on this?

Reply:
It works fine here. Don't forget to remove the

```
set_target_properties(libproxy PROPERTIES INTERFACE_LINK_LIBRARIES "")
set_target_properties(libproxy PROPERTIES LINK_INTERFACE_LIBRARIES "")
```

lines.

Reply:
Done. 
Thank you for reviewing